### ISSUE TITLE
After update to 9.1 can't add instances or edit the available ones

### ISSUE THREAD

**Issue by marcelovbcfilho** [PrismLauncher/PrismLauncher#2999]
### Operating System

Linux

### Version of Prism Launcher

9.1

### Version of Qt

Qt 6.7.3

### Description of bug

After updating to the newer 9.1 version I can't open the edit instance or add an instance. After clicking nothing happens and when I close prism all windows appear to open at once and close themselves.

### Steps to reproduce

Try and add an instance or edit an available one. I recorded a video showing the issue maybe this can help.

https://github.com/user-attachments/assets/2ac9f726-2681-4805-a408-28f16132b8c7

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by marcelovbcfilho** [comment]
I have done some extra testing and from what I can see it's probably a fetch related issue, in the following video when I have internet access the issue exists, but the internet is working properly, as you can see by me accessing the prismlauncher.org website, after I disable the internet and re-open the prism launcher I can edit and add instances, but now I'm without internet. Not sure, but maybe there is a lock with some parts of the launcher that await some sort of request that even I having internet access is not being finished which lead to the problem?

https://github.com/user-attachments/assets/83361e91-d531-466f-9bdb-e715ce36c1ec

**Comment by Trial97** [comment]
Please provide the logs from when this is happening(with internet) The logs can be found: `~/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/logs` (please upload them to https://mclo.gs/ or another pastebin service(also you may strip stuff that you are not comfortable to share).

From what I see in the videos, the versions task is stuck somehow. Can you access `https://meta.prismlauncher.org/v1/` in the browser? Can you check that the meta URL setting is empty(Settings>APIs>Metadata Server).

**Comment by marcelovbcfilho** [comment]
Ok, I uploaded the logs in the recommended site: https://mclo.gs/MUz1T6X and also recorded a video showing that I can access the api but the launcher still is locked somehow.
Maybe there is another lock not related to the prismlauncher api? If you have any other possible apis that might lock the launcher I can try them out.

https://github.com/user-attachments/assets/cdfc39ea-780d-49c3-8d9c-43fcb03f2fea

**Comment by Trial97** [comment]
Yep: it fails to load the versions.

Does this happens with other builds on your machine(e.g. appimage)?
Is the metadata server setting empty?
There was a recent fix regarding add instance button(to not wait for the load versions). Can you maybe try the nighlty builds and provide the logs from those?
https://nightly.link/PrismLauncher/PrismLauncher/workflows/trigger_builds/develop

**Comment by marcelovbcfilho** [comment]
In all cases the metadata server input was empty.

Here are the logs for the video when I test using official appimage, debug appimage(using the nightly.link provided) and the official fedora copr:

Official 9.1 AppImage: https://mclo.gs/aGFiaKf
Debug Build AppImage: https://mclo.gs/0Md2jWw
Fedora COPR Build: https://mclo.gs/m8XFadS

Also here is the video of me testing the 3 at once.

https://github.com/user-attachments/assets/bf9a84b9-a5fc-460b-87e5-27c79e88de7e

Edit: I tried typing something on the metadata server input but it seemed to be locked not allowing me to type anything to show that it was empty.

**Comment by Ryex** [comment]
could you try replicating the tests again but turn on network debug logging with 

```QT_LOGGING_RULES=launcher.task.net.download.debug=true``` set in the environment ?

**Comment by fptbb** [comment]
This is probably due to your ISP blocking GitHub Pages, please try to open and/or update https://meta.prismlauncher.org/v1/, it's also happening to a friend of mine. It would be great if prism provided some mirrors by default for this, like, Vercel and Cloudflare (I've created prism-meta.vercel.app myself and some other user already created prism-meta.pages.dev, so this must be a real problem even if it's not the case here).

**Comment by Frontear** [comment]
~~This is happening on Windows as well, using the MSVC portable builds. Pressing Edit will not open the window, however when I close the launcher I see a brief flicker of the screen before vanishing.~~

A reboot fixed it.. how disappointing. I was able to reproduce it 100% of the time with even a fresh instance but I guess it was some odd occurence on my system.

**Comment by marcelovbcfilho** [comment]
Sorry for not answering earlier, but here it is @Ryex a link for the logs using the provided environment variables: https://mclo.gs/CB7z3iE

Also here is a video @Fptbb showing that I can indeed access the provided link but the launcher still refuses to allow me to add a new instance.

In the video I'm using the Fedora COPR version to make the test.

https://github.com/user-attachments/assets/1f0fc416-321d-4619-b02b-9e7be882ab71

**Comment by fptbb** [comment]
> Also here is a video @Fptbb showing that I can indeed access the provided link but the launcher still refuses to allow me to add a new instance.

Indeed, @marcelovbcfilho, it seems that it is not your case, I read it and still commented just to warn other possible users searching for a solution in a similar issue.

**Comment by Ryex** [comment]
> Sorry for not answering earlier, but here it is @Ryex a link for the logs using the provided environment variables: https://mclo.gs/CB7z3iE

huh, that is sadly less than helpful...
it looks like it tries to access the meta servers and then just hangs...

to verify. you can still reproduce with a nightly build off the develop branch?
https://nightly.link/PrismLauncher/PrismLauncher/workflows/trigger_builds/develop


I wonder if github or some other middleman/ISP  is rate limiting or filtering prism's User Agent.
Can you try replacing the User Agent with something from a standard Browser?
eg. `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.53 Safari/537.36`

![image](https://github.com/user-attachments/assets/41bbb833-f787-4eca-b1ce-15a99ee4749d)

**Comment by TropheusJ** [comment]
Also experiencing this. Suddenly started happening on my home network, works fine on my college campus.
Still happens on the nightly build, even affects login interestingly.
![image](https://github.com/user-attachments/assets/678d6fc2-c8c2-4bc3-b328-931e82f633bb)

I can access the meta perfectly fine from my browser, changing prism's user agent had no effect.

**Comment by Ryex** [comment]
> Also experiencing this. Suddenly started happening on my home network, works fine on my college campus. Still happens on the nightly build, even affects login interestingly. 

when you say you can reproduce *this* on nightly. do you mean the windows refusing to open? or just the network issue?

**Comment by TropheusJ** [comment]
both, they're the same issue. The windows open fine once the requests finally make it through.

**Comment by marcelovbcfilho** [comment]
@Ryex Hey, sorry for the delay, I only have time to play on weekends. But here are the logs using the debug version you provided, I switched to windows this time but the issue still is the same: https://mclo.gs/2CxpgCd
Also recorded a video showing the issue even when changing the user agent of the launcher.

https://github.com/user-attachments/assets/bcc96bbb-4778-47e3-afe2-bfd2f5c42396

Just out of curiosity, I'm not developing the project, BUT, the request to verify if the launcher is using the latest version shouldn't be async? Allowing the user to edit their instances or add an instance even then the version verification fails?
